src/language/ast.js

```diff
@@ -625,3 +627,14 @@ export type InputObjectTypeExtensionNode = {|
   +directives?: $ReadOnlyArray<DirectiveNode>,
   +fields?: $ReadOnlyArray<InputValueDefinitionNode>,
 |};
+
+// Schema Coordinates
+
+export type SchemaCoordinateNode = {|
+  +kind: 'SchemaCoordinate',
+  +loc?: Location,
+  +isDirective: boolean,
+  +name: NameNode,
+  +fieldName?: NameNode,
```

Comment:
My main concern is enum values are accessed through the property `field` of the AST node.
I still think we need to have separate syntax for enum values.
I also have other arguments for separate syntax for non-fields, I will try to formulate them later.

Reply:
Yeah this is definitely a generalization and a tradeoff to keep this name short. What do you think about `fieldOrValueName` to make it clear that the one AST shape supports both forms?

Reply:
@leebyron I was thinking about something even more general like `subElementName`.
Since in future we can potentially make even more things addressable.